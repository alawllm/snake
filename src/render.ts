//does not modify variables
export const renderGameScreen = (
  ctx: CanvasRenderingContext2D,
  canvas: HTMLCanvasElement
): void => {
  ctx.fillStyle = "black";
  ctx.fillRect(0, 0, canvas.clientWidth, canvas.clientHeight);
};
//does not modify variables
export const drawGameOver = (
  ctx: CanvasRenderingContext2D,
  canvas: HTMLCanvasElement
): void => {
  ctx.fillStyle = "white";
  ctx.font = "55px handjet";
  ctx.fillText("Game Over!", canvas.clientWidth / 6.5, canvas.clientHeight / 2);
};
//does not modify variables
export const drawApple = (
  ctx: CanvasRenderingContext2D,
  appleX: number,
  appleY: number,
  tileCount: number,
  tileSize: number
): void => {
  ctx.fillStyle = "red";
  ctx.fillRect(appleX * tileCount, appleY * tileCount, tileSize, tileSize);
};
//renders score on screen - does not modify the score
export const setScoreOnScreen = (
  score: number,
  scoreContainer: HTMLElement
): void => {
  scoreContainer.textContent = "Score: " + score;
};
