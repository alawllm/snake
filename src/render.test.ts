import "jest-canvas-mock";
import { renderGameScreen } from "./render";

//this function paints black rectangle on the screen
// export const renderGameScreen = (
//     ctx: CanvasRenderingContext2D,
//     canvas: HTMLCanvasElement
//   ): void => {
//     ctx.fillStyle = "black";
//     ctx.fillRect(0, 0, canvas.clientWidth, canvas.clientHeight);
//   };

describe("draw", () => {
  const canvas = document.createElement("canvas") as HTMLCanvasElement;
  const ctx = canvas.getContext("2d") as CanvasRenderingContext2D;

  test("renders the game screen correctly", () => {
    renderGameScreen(ctx, canvas);
    expect(canvas).toMatchSnapshot();
  });
});
