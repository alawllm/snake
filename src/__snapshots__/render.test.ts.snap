// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`draw renders the game screen correctly 1`] = `<canvas />`;
